data/transactions/logic/assembler.go

```diff
@@ -1265,13 +1286,99 @@ func (ops *OpStream) checkStack(args StackTypes, returns StackTypes, instruction
 	}
 }
 
+const subRets int = -1
+const nowhere int = -2
+const exiting int = -3
+const erroring int = -4
+
+//A BasicBlock is a structure through which control can only flow in through start and out through end
+type BasicBlock struct {
+	startIndex       int         //index into ops.pending.Bytes()
+	endIndex         int         //Note this is the beginning of the last op in the block, somewhat unnecessary but we'll keep it for now
+	expectedStackTop []StackType //What the block requires to be on top of the stack in order to complete
+	stackDelta       []StackType //What the block net pushes to the stack
+	sourceErrors     []int       //Sourcelines for each entry in expectedStackTop which is useful for listing sourcelines for errors
+	jumpTo           int         //Index into ops.blocks, where control can jump to out of the block, i.e. via a branch
+	flowTo           int         //Where control can flow to, i.e. the block immediately following a conditional branch
```

Comment:
I think I expected *Block for these.  Convince me that they should be named by indexes.

Reply:
Real answer: It feels somewhat more flexible when coding to use indices since it makes it easier for example to say flowTo is len(blocks)+1. This gives you info that you can use while you're still creating blocks instead of waiting until after; I can't recall ever really needing that info in any of the versions of the typechecker but it feels nice to have. Also just a note, if we keep them as is, we can change fixJumpsAndFlows to only fix jumps since the flowTo's will already be correct. 

Fake answer: 4 bytes per index saved